RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitPhase2WeightsProducer.cc

```diff
@@ -0,0 +1,111 @@
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+
+#include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
+#include "CondFormats/EcalObjects/interface/EcalCATIAGainRatios.h"
+#include "CondFormats/DataRecord/interface/EcalCATIAGainRatiosRcd.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
+#include "DataFormats/EcalRecHit/interface/EcalUncalibratedRecHit.h"
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+
+#include <TGraph.h>
+#include <TF1.h>
+
+class EcalUncalibRecHitPhase2WeightsProducer: public edm::stream::EDProducer<>{
+
+
+public:
+  explicit EcalUncalibRecHitPhase2WeightsProducer(const edm::ParameterSet& ps);//, edm::ConsumesCollector& c);
+  void produce(edm::Event& evt, const edm::EventSetup& es) override;
+
+private:
+  
+  float tRise_;
+  float tFall_;  
+  double gainRatios[2]={10,1};  
+  float gratio;
```

Comment:
This variable does not look as if it need to be a class member. It is only used once in `produce`.